js/siteorigin-panels/dialog/widget.js

```diff
@@ -274,16 +274,16 @@ module.exports = panels.view.dialog.extend( {
 	 * @returns {boolean}
 	 */
 	deleteHandler: function () {
-
-		this.model.trigger( 'visual_destroy' );
+		this.widgetView.visualDestroyModel();
 		this.closeDialog( {silent: true} );
 		this.builder.model.refreshPanelsData();
 
 		return false;
 	},
 
 	duplicateHandler: function () {
-		this.model.trigger( 'user_duplicate' );
+		// Call the widget duplicate handler directly
+		this.widgetView.duplicateHandler();
```

Comment:
Views shouldn't be reaching outside themselves and calling functions on other views, but yeah, then you'd have to duplicate the duplication handling code here. 😅 I guess this is one of the situations where a controller class would come in handy.

Reply:
Possible we can just call the `this.widgetView.duplicateHandler` directly from events in the dialog. Not sure if that's better practice?